### PR TITLE
Jwt token

### PR DESCRIPTION
**Pull request by JohnGrey17**

### CODE REVIEW

File: src/main/java/org/example/bookstore/dto/userdto/UserLoginRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package org.example.bookstore.dto.userdto;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import org.hibernate.validator.constraints.Length;
+
+public record UserLoginRequestDto(
+        @NotBlank
+        @Email
+        String email,
```
Comment (d1sam) [inline src/main/java/org/example/bookstore/dto/userdto/UserLoginRequestDto.java:11]: what about the size of an email?

File: src/main/java/org/example/bookstore/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@RequiredArgsConstructor
+
```
Comment (d1sam) [inline src/main/java/org/example/bookstore/security/JwtAuthenticationFilter.java]:
```suggestion
```

File: src/main/java/org/example/bookstore/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer")) {
+            return bearerToken.substring(7);
+        }
+        return null;
```
Comment (d1sam) [inline src/main/java/org/example/bookstore/security/JwtAuthenticationFilter.java]: please avoid magic numbers and string literals

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -9,3 +9,6 @@ spring.jpa.show-sql=true
 spring.jpa.open-in-view=false
 spring.jackson.deserialization.fail-on-unknown-properties=true
 server.servlet.context-path=/api
+
+jwt.expiration=300000
+jwt.secret=bigmateacedemysecretfeb24blablablablablaoiuluzichervonakaluna`
```
Comment (d1sam) [inline src/main/resources/application.properties]: let's provide these values from env properties

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, but there are some minor changes, that should be done according my comments.

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**